File: webdataos-v2/sdks/typescript/examples/basic-usage.ts
import { WebDataGatewayClient } from "../src";

async function main() {
  const client = new WebDataGatewayClient({ baseUrl: "http://localhost:8000" });
  const result = await client.fetch({
    url: "https://example.com/pricing",
    task_type: "pricing_extraction",
    preferred_tool: "web_scraper_api",
    output_schema: {
      company: "string",
      pricing_model: "string",
      features: "list",
    },
  });
  console.log(result.status, result.tool_used);
  console.log(result.data);
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: webdataos-v2/sdks/python/examples/basic_usage.py
from webdata_gateway import GatewayFetchRequest, WebDataGatewayClient


def main() -> None:
    with WebDataGatewayClient() as client:
        response = client.fetch(
            GatewayFetchRequest(
                url="https://example.com/pricing",
                task_type="pricing_extraction",
                preferred_tool="web_scraper_api",
                output_schema={
                    "company": "string",
                    "pricing_model": "string",
                    "features": "list",
                },
            )
        )
        print(response.status)
        print(response.tool_used)
        print(response.data)


if __name__ == "__main__":
    main()
